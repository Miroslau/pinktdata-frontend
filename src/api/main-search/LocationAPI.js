import httpClient from '../index';

export default {
  async search(location) {
    const res = await httpClient.get(`/apartments/locations?query=${location}`);
    const { data } = res;
    return data;
  },
};
