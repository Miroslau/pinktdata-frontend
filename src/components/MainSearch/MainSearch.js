import React, { useState } from 'react';
import { Button, TextField } from '@mui/material';
import AdapterDateFns from '@mui/lab/AdapterDateFns';
import FormControl from '@mui/material/FormControl';
import DatePicker from '@mui/lab/DatePicker';
import LocalizationProvider from '@mui/lab/LocalizationProvider';
import Select from '@mui/material/Select';
import MenuItem from '@mui/material/MenuItem';
import InputLabel from '@mui/material/InputLabel';
import './MainSearch.scss';

const MainSearch = () => {
  const [location, setLocation] = useState('');
  const [startDateValue, setStartDateValue] = useState(new Date('2021-10-10T11:11:11'));
  const [endDateValue, setEndDateValue] = useState(new Date('2021-12-10T11:11:11'));

  const [bedroomValue, setBedroomValue] = useState('');

  const handleBedroomValue = (event) => {
    setBedroomValue(event.target.value);
  };

  return (
    <form
      className="searchForm"
      sx={{
        width: '65%',
        display: 'flex',
        padding: '2rem',
        justifyContent: 'space-between',
        backgroundColor: '#fff',
        boxShadow: '1px 1px 25px 13px rgba(34, 60, 80, 0.14)',
        borderRadius: '1rem',
        margin: '0 auto',
      }}
    >
      <TextField
        sx={{
          '& .MuiInput-root': {
            '& MuiInput-root input:before': {
              border: 'none',
            },
          },
        }}
        label="Location"
        variant="standard"
        placeholder="Enter location"
        value={location}
        onChange={(event) => {
          setLocation(event.target.value);
        }}
      />
      <LocalizationProvider dateAdapter={AdapterDateFns}>
        <DatePicker
          label="Start Date"
          value={startDateValue}
          onChange={setStartDateValue}
          renderInput={(params) => <TextField {...params} />}
        />

        <DatePicker
          label="End Date"
          value={endDateValue}
          onChange={setEndDateValue}
          min={startDateValue}
          renderInput={(params) => <TextField {...params} />}
        />
      </LocalizationProvider>

      <FormControl sx={{ width: 250 }}>
        <InputLabel id="bedroomValue">Bedroom type</InputLabel>
        <Select
          labelId="bedroomValue"
          value={bedroomValue}
          label="Bedroom type"
          onChange={handleBedroomValue}
        >
          <MenuItem value={1}>One</MenuItem>
          <MenuItem value={2}>Two</MenuItem>
          <MenuItem value={3}>Three</MenuItem>
        </Select>
      </FormControl>

      <Button variant="contained" color="secondary" type="submit">Search</Button>

    </form>
  );
};

export default MainSearch;
