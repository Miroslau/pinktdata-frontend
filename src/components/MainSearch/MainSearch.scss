.searchForm {
  width: 65%;
  display: flex;
  padding: 2rem;
  justify-content: space-between;
  background-color: #fff;
  box-shadow: 1px 1px 25px 13px rgba(34, 60, 80, 0.14);
  border-radius: 1rem;
  margin: 0 auto;
}

input::before, input::after {
  border: none;
}
