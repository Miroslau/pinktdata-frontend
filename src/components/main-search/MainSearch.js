import React, { useState } from 'react';
import { TextField } from '@mui/material';
import AdapterDateFns from '@mui/lab/AdapterDateFns';
import FormControl from '@mui/material/FormControl';
import DatePicker from '@mui/lab/DatePicker';
import LocalizationProvider from '@mui/lab/LocalizationProvider';
import Select from '@mui/material/Select';
import MenuItem from '@mui/material/MenuItem';
import InputLabel from '@mui/material/InputLabel';
import RoomIcon from '@mui/icons-material/Room';
import SearchIcon from '@mui/icons-material/Search';
import useStyles from './MainSearch.style';
import ButtonComponent from '../button-component/ButtonComponent';

const bedroomItems = [
  {
    id: 1,
    value: 1,
    title: 'One',
  },
  {
    id: 2,
    value: 2,
    title: 'Two',
  },
  {
    id: 3,
    value: 3,
    title: 'Three',
  },
];

const MainSearch = () => {
  const classes = useStyles();
  const [location, setLocation] = useState('');
  const [startDateValue, setStartDateValue] = useState(new Date('2021-10-10T11:11:11'));
  const [endDateValue, setEndDateValue] = useState(new Date('2021-12-10T11:11:11'));

  const [bedroomValue, setBedroomValue] = useState('');

  const handleBedroomValue = (event) => {
    setBedroomValue(event.target.value);
  };

  return (
    <form className={classes.form}>
      <div className={classes.locationWrapper}>
        <RoomIcon className={classes.roomIcon} />
        <TextField
          className={classes.location}
          label="Location"
          variant="standard"
          placeholder="Enter location"
          value={location}
          onChange={(event) => {
            setLocation(event.target.value);
          }}
        />
      </div>

      <LocalizationProvider dateAdapter={AdapterDateFns}>
        <DatePicker
          label="Start Date"
          value={startDateValue}
          onChange={setStartDateValue}
          renderInput={(params) => <TextField {...params} className={classes.date} />}
        />

        <DatePicker
          label="End Date"
          value={endDateValue}
          onChange={setEndDateValue}
          min={startDateValue}
          renderInput={(params) => <TextField {...params} className={classes.date} />}
        />
      </LocalizationProvider>

      <FormControl className={classes.bedroom}>
        <InputLabel id="bedroomValue">Bedroom type</InputLabel>
        <Select
          labelId="bedroomValue"
          value={bedroomValue}
          label="Bedroom type"
          onChange={handleBedroomValue}
        >

          {bedroomItems.map(({ id, value, title }) => (
            <MenuItem
              key={id}
              value={value}
            >
              {title}
            </MenuItem>
          ))}
        </Select>
      </FormControl>

      <ButtonComponent variant="contained" color="secondary" type="submit" className={classes.searchButton}>
        <SearchIcon fontSize="large" />
      </ButtonComponent>

    </form>
  );
};

export default MainSearch;
