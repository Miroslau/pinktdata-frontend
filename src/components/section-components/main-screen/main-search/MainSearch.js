import React, { useState } from 'react';
import { TextField } from '@mui/material';
import AdapterDateFns from '@mui/lab/AdapterDateFns';
import FormControl from '@mui/material/FormControl';
import DatePicker from '@mui/lab/DatePicker';
import LocalizationProvider from '@mui/lab/LocalizationProvider';
import Select from '@mui/material/Select';
import MenuItem from '@mui/material/MenuItem';
import InputLabel from '@mui/material/InputLabel';
import RoomIcon from '@mui/icons-material/Room';
import SearchIcon from '@mui/icons-material/Search';
import ButtonMui from '../../../ui-components/button-mui/ButtonMui';
import TextFieldMui from '../../../ui-components/text-field-mui/TextFieldMui';
import TEXT from '../../../../constants/mainScreen';
import searchByLocation from '../../../../api/main-search/main-searchAPI';
import useStyles from './MainSearch.style';

const bedroomItems = [
  {
    id: 1,
    value: 1,
    title: 'One',
  },
  {
    id: 2,
    value: 2,
    title: 'Two',
  },
  {
    id: 3,
    value: 3,
    title: 'Three',
  },
];

const MainSearch = () => {
  const dateNow = new Date();
  const dateNowPlusOneDay = new Date();
  dateNowPlusOneDay.setDate(dateNow.getDate() + 1);
  dateNowPlusOneDay.toLocaleDateString();

  const classes = useStyles();
  const [location, setLocation] = useState('');
  const [startDateValue, setStartDateValue] = useState(dateNow);
  const [endDateValue, setEndDateValue] = useState(dateNowPlusOneDay);

  const [bedroomValue, setBedroomValue] = useState('');

  const handleBedroomValue = (event) => {
    setBedroomValue(event.target.value);
  };

  return (
    <form className={classes.form}>
      <div className={classes.locationWrapper}>
        <RoomIcon className={classes.roomIcon} />
        <TextFieldMui
          className={classes.location}
          label={TEXT.MAIN_SEARCH.LOCATION}
          variant="standard"
          placeholder={TEXT.MAIN_SEARCH.ENTER_LOCATION}
          value={location}
          onChange={(event) => {
            setLocation(event.target.value);
            searchByLocation.location();
          }}
        />
      </div>

      <LocalizationProvider dateAdapter={AdapterDateFns}>
        <DatePicker
          label={TEXT.MAIN_SEARCH.START_DATE}
          value={startDateValue}
          onChange={setStartDateValue}
          renderInput={(params) => <TextField {...params} className={classes.date} />}
        />

        <DatePicker
          label={TEXT.MAIN_SEARCH.END_DATE}
          value={endDateValue}
          onChange={setEndDateValue}
          min={startDateValue}
          renderInput={(params) => <TextField {...params} className={classes.date} />}
        />
      </LocalizationProvider>

      <FormControl className={classes.bedroom}>
        <InputLabel id="bedroomValue">{TEXT.MAIN_SEARCH.BEDROOM_TYPE}</InputLabel>
        <Select
          labelId="bedroomValue"
          value={bedroomValue}
          label={TEXT.MAIN_SEARCH.BEDROOM_TYPE}
          onChange={handleBedroomValue}
        >

          {bedroomItems.map(({ id, value, title }) => (
            <MenuItem
              key={id}
              value={value}
            >
              {title}
            </MenuItem>
          ))}
        </Select>
      </FormControl>

      <ButtonMui ariaLabel="search-button" variant="contained" color="secondary" type="submit" className={classes.searchButton}>
        <SearchIcon fontSize="large" />
      </ButtonMui>

    </form>
  );
};

export default MainSearch;
