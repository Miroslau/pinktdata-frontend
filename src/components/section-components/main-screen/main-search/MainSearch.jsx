import React, { useEffect, useRef, useState } from 'react';
import { TextField } from '@mui/material';
import Autocomplete from '@mui/material/Autocomplete';
import AdapterDateFns from '@mui/lab/AdapterDateFns';
import FormControl from '@mui/material/FormControl';
import DatePicker from '@mui/lab/DatePicker';
import LocalizationProvider from '@mui/lab/LocalizationProvider';
import Select from '@mui/material/Select';
import MenuItem from '@mui/material/MenuItem';
import InputLabel from '@mui/material/InputLabel';
import RoomIcon from '@mui/icons-material/Room';
import SearchIcon from '@mui/icons-material/Search';
import debounce from 'lodash.debounce';
import ButtonMui from '../../../ui-components/button-mui/ButtonMui';
import TEXT from '../../../../constants/mainScreen';
import useStyles from './MainSearch.style';
import LocationAPI from '../../../../api/main-search/LocationAPI';

const bedroomItems = [
  {
    id: 1,
    value: 1,
    title: 'One',
  },
  {
    id: 2,
    value: 2,
    title: 'Two',
  },
  {
    id: 3,
    value: 3,
    title: 'Three',
  },
];

const MainSearch = () => {
  const dateNow = new Date();
  const dateNowPlusOneDay = new Date();
  dateNowPlusOneDay.setDate(dateNow.getDate() + 1);
  dateNowPlusOneDay.toLocaleDateString();

  const classes = useStyles();
  const isMounted = useRef(null);
  const [location, setLocation] = useState('');
  const [dataLocation, setDataLocation] = useState([]);
  const [startDateValue, setStartDateValue] = useState(dateNow);
  const [endDateValue, setEndDateValue] = useState(dateNowPlusOneDay);
  const [bedroomValue, setBedroomValue] = useState('');
  const [isError, setIsError] = useState(false);

  const defaultProps = {
    options: dataLocation,
  };

  useEffect(() => {
    isMounted.current = true;
    return () => {
      isMounted.current = false;
    };
  });

  useEffect(async () => {
    if (location) {
      try {
        const response = await LocationAPI.search(location);
        if (isMounted.current) {
          console.log(response.cities);
          setDataLocation(response.cities);
        }
      } catch (error) {
        console.log(error);
      }
    }
  }, [location]);

  const handleBedroomValue = (event) => {
    setBedroomValue(event.target.value);
  };

  const handleChange = (e) => {
    const { value } = e.target;
    if (!value) {
      setDataLocation([]);
      setIsError(true);
    } else {
      setIsError(false);
      setLocation(value);
    }
  };
  const searchChangeHandler = debounce(handleChange, 500);

  return (
    <form className={classes.form}>
      <div className={classes.locationWrapper}>
        <RoomIcon className={classes.roomIcon} />
        <Autocomplete
          style={{ width: '300px' }}
          {...defaultProps}
          id="disable-close-on-select"
          disableClearable
          renderInput={(params) => (
            <TextField
              data-testid="Location"
              placeholder="Location"
              {...params}
              onChange={searchChangeHandler}
              value={location}
              label={TEXT.MAIN_SEARCH.LOCATION}
              variant="standard"
              error={isError}
            />
          )}
        />
      </div>

      <LocalizationProvider dateAdapter={AdapterDateFns}>
        <DatePicker
          label={TEXT.MAIN_SEARCH.START_DATE}
          value={startDateValue}
          onChange={setStartDateValue}
          renderInput={(params) => <TextField {...params} className={classes.date} />}
        />

        <DatePicker
          label={TEXT.MAIN_SEARCH.END_DATE}
          value={endDateValue}
          onChange={setEndDateValue}
          min={startDateValue}
          renderInput={(params) => <TextField {...params} className={classes.date} />}
        />
      </LocalizationProvider>

      <FormControl className={classes.bedroom}>
        <InputLabel id="bedroomValue">{TEXT.MAIN_SEARCH.BEDROOM_TYPE}</InputLabel>
        <Select
          labelId="bedroomValue"
          value={bedroomValue}
          label={TEXT.MAIN_SEARCH.BEDROOM_TYPE}
          onChange={handleBedroomValue}
        >

          {bedroomItems.map(({ id, value, title }) => (
            <MenuItem
              key={id}
              value={value}
            >
              {title}
            </MenuItem>
          ))}
        </Select>
      </FormControl>

      <ButtonMui
        ariaLabel="search-button"
        variant="contained"
        color="secondary"
        className={classes.searchButton}
        clickButton={() => console.log('Search...')}
      >
        <SearchIcon fontSize="large" />
      </ButtonMui>

    </form>
  );
};

export default MainSearch;
