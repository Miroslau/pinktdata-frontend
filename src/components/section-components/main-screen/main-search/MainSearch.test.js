/* eslint-disable no-unused-vars */
/* eslint-disable react/jsx-one-expression-per-line */
/* eslint-disable max-len */
import '@testing-library/jest-dom';
import {
  render, fireEvent, act, screen,
} from '@testing-library/react';
import { createMemoryHistory } from 'history';
import { BrowserRouter, Router } from 'react-router-dom';
import LocationAPI from '../../../../api/main-search/LocationAPI';
import MainSearch from './MainSearch';

const mockDispatch = jest.fn();
jest.mock('react-redux', () => ({
  useDispatch: () => mockDispatch,
}));

jest.mock('../../../../api/main-search/LocationAPI');

jest.mock('../../../../utils/doWithUserDelay', () => ({
  doWithUserDelay: (callback) => callback(),
}));

describe('MainSearch component', () => {
  const setup = () => {
    LocationAPI.search = jest.fn().mockReturnValue(['Tafton, PA, United States',
      'Taghkanic, NY, United States',
      'Takoma Park, MD, United States']);

    const history = createMemoryHistory({ initialEntries: ['/'] });
    const utils = render(<Router location={history.location} navigator={history}><MainSearch /></Router>);
    const autocomplete = utils.getByTestId('autocomplete');
    const input = utils.getByPlaceholderText('Location');
    const button = utils.getByTestId('search-button');

    const selectOption = async () => {
      await act(async () => {
        autocomplete.focus();
        fireEvent.change(input, { target: { value: 't' } });
      });
      fireEvent.keyDown(autocomplete, { key: 'ArrowDown' });
      fireEvent.keyDown(autocomplete, { key: 'Enter' });
    };

    return {
      button,
      autocomplete,
      input,
      history,
      selectOption,
      ...utils,
    };
  };

  it('should render Search form', () => {
    const { container } = setup();
    expect(container.firstChild).toBeInTheDocument();
  });

  it('should render disabled button by default', () => {
    const { button } = setup();
    expect(button).toBeDisabled();
  });

  it('should select autocomplete option', async () => {
    const {
      input, selectOption,
    } = setup();
    await selectOption();
    expect(input.value).toBe('Taghkanic, NY, United States');
  });

  it('should activate button when option selected', async () => {
    const {
      button, selectOption,
    } = setup();
    await selectOption();
    expect(button).not.toBeDisabled();
  });

  it('should forward to map page when button clicked', async () => {
    const {
      button, selectOption, history,
    } = setup();
    await selectOption();
    fireEvent.click(button);
    expect(history.location.pathname).toBe('/map');
  });

  it('should select first option on blur', async () => {
    const { input } = setup();
    await act(async () => {
      input.focus();
      fireEvent.change(input, { target: { value: 't' } });
      expect(LocationAPI.search).toHaveBeenCalledTimes(1);
    });
    input.blur();
    expect(input.value).toBe('Tafton, PA, United States');
  });

  it('should highlight input when no value', async () => {
    const { input, container } = setup();
    const label = container.querySelector('#disable-close-on-select-label');
    await act(async () => {
      input.focus();
      fireEvent.change(input, { target: { value: '' } });
      input.blur();
    });
    expect(input.value).toBe('');
    expect(label.classList.contains('Mui-error')).toBe(true);
  });

  test('renders component bedroom count', () => {
    const history = createMemoryHistory();
    render(<Router location={history.location} navigator={history}><MainSearch /></Router>);
    expect(screen.getByText('Bedroom count')).toBeInTheDocument();
  });

  test('has open popup when click on label', async () => {
    const history = createMemoryHistory();
    render(<Router location={history.location} navigator={history}><MainSearch /></Router>);
    const button = screen.getByText(/bedroom count/i);
    fireEvent.click(button);
    const popupText = await screen.findByText('Select the number of rooms');
    expect(popupText).toBeInTheDocument();
  });

  test('has change count bedrooms when click button plus', async () => {
    const history = createMemoryHistory();
    render(<Router location={history.location} navigator={history}><MainSearch /></Router>);
    const textLabel = await screen.findByText(/Bedroom count/i);
    fireEvent.click(textLabel);
    const buttonAdd = await screen.getByRole('button', { name: /add/i });
    fireEvent.click(buttonAdd);
    const textLabelAdd = await screen.findByText(/Bedroom count: 1/i);
    expect(textLabelAdd).toBeInTheDocument();
  });

  test('has change count bedrooms when click button minus', async () => {
    const history = createMemoryHistory();
    render(<Router location={history.location} navigator={history}><MainSearch /></Router>);
    const textLabel = await screen.findByText(/Bedroom count/i);
    fireEvent.click(textLabel);
    const buttonAdd = await screen.getByRole('button', { name: /add/i });
    const countClick = [...Array(2)];
    countClick.forEach(() => fireEvent.click(buttonAdd));
    const buttonRemove = await screen.getByRole('button', { name: /remove/i });
    fireEvent.click(buttonRemove);
    const textLabelAdd = await screen.findByText(/Bedroom count: 1/i);
    expect(textLabelAdd).toBeInTheDocument();
  });

  test('has not change count bedrooms when click the button plus if bedrooms more 8', async () => {
    const history = createMemoryHistory();
    render(<Router location={history.location} navigator={history}><MainSearch /></Router>);
    const textLabel = await screen.findByText(/Bedroom count/i);
    fireEvent.click(textLabel);
    const buttonAdd = await screen.getByRole('button', { name: /add/i });
    const countClick = [...Array(9)];
    countClick.forEach(() => fireEvent.click(buttonAdd));
    const textLabelAdd = await screen.findByText(/Bedroom count: 8/i);
    expect(textLabelAdd).toBeInTheDocument();
  });

  test('has not change count bedrooms when click the button minus if bedrooms 0', async () => {
    const history = createMemoryHistory();
    render(<Router location={history.location} navigator={history}><MainSearch /></Router>);
    const textLabel = await screen.findByText(/Bedroom count/i);
    fireEvent.click(textLabel);
    const buttonRemove = await screen.getByRole('button', { name: /remove/i });
    const countClick = [...Array(9)];
    countClick.forEach(() => fireEvent.click(buttonRemove));
    const textLabelAdd = await screen.findByText(/0/i);
    expect(textLabelAdd).toBeInTheDocument();
  });
});
