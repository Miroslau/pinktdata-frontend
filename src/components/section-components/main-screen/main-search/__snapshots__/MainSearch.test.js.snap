// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`calls "onClick" prop on button click 1`] = `
<DocumentFragment>
  <form
    class="makeStyles-form-17"
  >
    <div
      class="makeStyles-locationWrapper-18"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium makeStyles-roomIcon-22 css-i4bv87-MuiSvgIcon-root"
        data-testid="RoomIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 2C8.13 2 5 5.13 5 9c0 5.25 7 13 7 13s7-7.75 7-13c0-3.87-3.13-7-7-7zm0 9.5c-1.38 0-2.5-1.12-2.5-2.5s1.12-2.5 2.5-2.5 2.5 1.12 2.5 2.5-1.12 2.5-2.5 2.5z"
        />
      </svg>
      <div
        class="MuiFormControl-root MuiTextField-root makeStyles-location-19 css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
      >
        <label
          class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard MuiFormLabel-root MuiFormLabel-colorPrimary css-aqpgxn-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="false"
        >
          Location
        </label>
        <div
          class="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-1480iag-MuiInputBase-root-MuiInput-root"
        >
          <input
            aria-invalid="false"
            class="MuiInput-input MuiInputBase-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
            placeholder="Enter location"
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
    <div
      class="MuiFormControl-root MuiTextField-root makeStyles-date-20 css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
    >
      <label
        class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary MuiFormLabel-filled css-1kty9di-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink="true"
      >
        Start Date
      </label>
      <div
        class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
      >
        <input
          aria-invalid="false"
          aria-label="Choose date, selected date is Oct 30, 2021"
          aria-readonly="true"
          class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
          readonly=""
          type="text"
          value="10/30/2021"
        />
        <fieldset
          aria-hidden="true"
          class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="css-186xcr5"
          >
            <span>
              Start Date
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
    <div
      class="MuiFormControl-root MuiTextField-root makeStyles-date-20 css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
    >
      <label
        class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary MuiFormLabel-filled css-1kty9di-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink="true"
      >
        End Date
      </label>
      <div
        class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
      >
        <input
          aria-invalid="false"
          aria-label="Choose date, selected date is Oct 31, 2021"
          aria-readonly="true"
          class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
          readonly=""
          type="text"
          value="10/31/2021"
        />
        <fieldset
          aria-hidden="true"
          class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="css-186xcr5"
          >
            <span>
              End Date
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
    <div
      class="MuiFormControl-root makeStyles-bedroom-21 css-1nrlq1o-MuiFormControl-root"
    >
      <div
        class="MuiFormControl-root css-5jdat-MuiFormControl-root"
      >
        <label
          class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-contained MuiFormLabel-root MuiFormLabel-colorPrimary css-aqpgxn-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="false"
          id="input-search-bedrooms"
        >
          Bedroom count
        </label>
      </div>
    </div>
    <button
      aria-label="search-button"
      class="MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root makeStyles-searchButton-23 css-zcbmsk-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge css-tzssek-MuiSvgIcon-root"
        data-testid="SearchIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
        />
      </svg>
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </form>
</DocumentFragment>
`;
